src/ResourceManagement/ResourceManager/Authentication/AzureCredentials.cs

```diff
@@ -175,8 +176,25 @@ public async override Task ProcessHttpRequestAsync(HttpRequestMessage request, C
 #else
                     else if (servicePrincipalLoginInformation.X509Certificate != null)
                     {
+                        IdentityModel.Clients.ActiveDirectory.ClientAssertionCertificate clientAssertionCertificate = 
+                            new IdentityModel.Clients.ActiveDirectory.ClientAssertionCertificate(servicePrincipalLoginInformation.ClientId, servicePrincipalLoginInformation.X509Certificate);
+                        CertificateAuthenticationProvider certificateAuthenticationProvider = 
+                            new CertificateAuthenticationProvider(clientAssertionCertificate, servicePrincipalLoginInformation.IsCertificateRollOverEnabled);
+
+                        credentialsCache[adSettings.TokenAudience] = await ApplicationTokenProvider.LoginSilentAsync(
```

Comment:
not sure the logic, why `certificateAuthenticationProvider` is a must now with CertRollOverEnabled

Reply:
It was what the issue needed. The default `false` is not breaking with current logic.